backends/dpdk/dpdkAsmOpt.h

```diff
@@ -173,6 +173,125 @@ class ValidateTableKeys : public Inspector {
     int getFieldSizeBits(const IR::Type *field_type);
 };
 
+// This pass shorten the Identifier length
+class ShortenTokenLength : public Transform {
+    ordered_map<cstring, cstring> newNameMap;
+    ordered_map<cstring, cstring> &origNameMap;
+    static size_t count;
+    // Currently Dpdk allows Identifier of 63 char long or less
+    // including dots(.) for member exp.
+    // worst case member expression will look like below(for headers)
+    // 1.30.30 => 63(including dot(.))
+    cstring shortenString(cstring str, size_t allowedLength = 60) {
```

Comment:
We already have a class which does this, MinimalNameGenerator.
You should reuse that code.

Reply:
This runs post codegen and applied on some dpdk specific types too.